libs/queue/SPECIFICATION.md

```diff
@@ -4,32 +4,34 @@ This document provides an overview of the Queue library.
 
 It outlines the use cases, i.e. specification, and describes how to implement the library.
 
+The `Queue<T>` is a linear structure with a generic type parameter `T`. The elements in the queue are stored as a `Vec<T>`. Queues are growable, operate in a First-In-First-Out (FIFO) manner, and are created empty.
+
 ## Use Cases
 
-The Queue library can be used as a traditional queue or first in first out data structure.
+The Queue library can be used First-In-First-Out data structure when you want to get items out in the order that they are added.
 
 ## Public Functions
 
 ### `new()`
 
-Creates a new instance of a `Queue`. 
+- Creates a new instance of a `Queue`. 
 
 ### `is_empty()`
 
-Returns a boolean indicating whether the length of the `Queue` is zero. 
+- Returns a boolean indicating whether the length of the `Queue` is zero. 
 
 ### `len()`
 
-Returns the number of elements in the `Queue`
+- Returns the number of elements in the `Queue`
 
 ### `enqueue()`
 
-Adds the passed element to the `Queue`
+- Adds the item into the `Queue`
```

Comment:
As per my previous comment, if you're going to go with the suggestion of an `item` then every instance of the word `element` in the readme and spec should also be updated otherwise we should consistently use `element`